src/main/java/io/aiven/kafka/connect/transforms/Hash.java

```diff
@@ -88,7 +88,19 @@ public R apply(final R record) {
         }
 
         if (newValue.isPresent()) {
-            return record.newRecord(
+            if (dataPlace().equals("key")) {
```

Comment:
Probably, instead of conditioning here, it'd better to move specific logic into subclasses (in the test as well).

Reply:
Gotcha. I can do that for his one and the concat transformer

Reply:
Ok I've updated with that change!